docs/_posts/testing_a_tricky_edge_case_in_a_tuf_client.md

```diff
@@ -0,0 +1,117 @@
+{
+ "signatures": [{...}],
+ "signed": {
+  "_type": "timestamp",
+  <b>"expires": "2022-04-13T00:18:31Z"</b>,
+  "meta": { "snapshot.json": {"version": 2}},
+  "spec_version": "1.0.28",
+  "version": 2
+ }}
+</code></pre>
+
+As you can see, the first date is 30 Mar and the second - 13 Apr, which is exactly 14 days later. This is a great way to observe what the tests really do and check if they do it successfully.
+
+## When We Talk about Security, Edge Cases are the Norm
+
+Now, let’s take a closer look at two edge cases, using in this test the two cool things the RepositorySimulator provides:
+
+The TUF spec says the following:
+
+“Record the time at which the update began as the fixed update start time. Time is fixed at the beginning of the update workflow to allow an application using TUF to effectively pause time, in order to ensure that metadata which has a valid expiration time at the beginning of an update does not fail an expiration check later in the update workflow.” Let's see the following examples where this is not so obvious.
+
+Let's consider the following examples where this is not so obvious.
+
+### Example with expired metadata:
+
+Imagine that we have performed an update and stored metadata in a cache. And the locally stored timestamp/snapshot has expired. But we still need it to perform an update from remote by verifying the signatures and we need to use the expired timestamp. Here the repository simulator can do a great job testing this scenario, as we’d like to somehow update expiry without explicitly touching the file system from the tests.
+
+We can play with versions and expiry to verify that this scenario not explicitly mentioned in the spec works correctly and safely. By using the simulator, we can do the following:
+1. Set the timestamp expiry one week ahead (Day 7)
+2. On the very first day download, verify, and load metadata for the [top-level roles](https://theupdateframework.github.io/specification/latest/#roles-and-pki) following the TUF specification order. This is done by simply calling `RespositorySimulator.refresh()`.
+3. Then we bump [snapshot](https://theupdateframework.github.io/specification/latest/#update-snapshot) and [targets](https://theupdateframework.github.io/specification/latest/#targets) versions to v2 on the same day
+4. Set v2 expiry dates three weeks ahead (Day 21)
+5. Travel in time somewhere between Day 7 and Day 21
+6. Perform a successful `refresh` (with `RepositorySimulator.refresh()` call) with the expired locally cached timestamp
+7. Check that the final version of the snapshot and targets roles is v2.
+This is a not so obvious use-case to keep in mind when thinking about updates. You can see how it looks in practice in the [reference implementation](https://github.com/theupdateframework/python-tuf/blob/develop/tests/test_updater_top_level_update.py#:~:text=test_expired_metadata).
```

Comment:
I must say, the docstring form `test_expired_metadata` describes this chain of events slightly clearer:
- it mentions what happens in the repo and what happens on the client
- it mentions for each event when it happens in a uniform way

Reply:
Do you think it's better to cite the docstring, or link to it, or cite and elaborate?

Reply:
Cite and elaborate probably. :) I just wanted to point out that currently it is not immediately obvious if something happens in the repo or on the client. If you polish that a bit, I'm happy. My second comment about the time is less important.